addons/website_slides/controllers/main.py

```diff
@@ -731,16 +731,29 @@ def slide_quiz_submit(self, slide_id, answer_ids):
 
         rank_progress = {}
         if not user_bad_answers:
+            old_lower_bound = request.env.user.rank_id.karma_min
```

Comment:
This is missing a little bit of structure.

I would suggest:

```
rank_progress['previous_rank'] = self._get_rank_values(request.env.user)
slide._action_set_quiz_done()
slide.action_set_completed()
rank_progress['new_rank'] = self._get_rank_values(request.env.user)
rank_progress.update({
    'description': request.env.user.rank_id.description,
    'last_rank': not request.env.user.next_rank_id,
    'level_up': rank_progress['previous_rank']['lower_bound'] != rank_progress['new_rank']['lower_bound'],
})

...

def _get_rank_values(self, user):
    lower_bound = user.rank_id.karma_min
    upper_bound = user.next_rank_id.karma_min
    return {
	'lower_bound': lower_bound,
	'upper_bound': upper_bound,
	'karma': user.karma,
	'motivational': user.next_rank_id.description_motivational,
        'progress': 100 if not user.next_rank_id else 100 * ((user.karma - lower_bound) / (upper_bound - lower_bound))
    }
```

Of course you would also have to adapt the JS accordingly.